### ISSUE TITLE
DK: Death Knights shoud be ressurected in bg with full runic power

### ISSUE THREAD

**Issue by CraftedRO** [TrinityCore/TrinityCore#29066]
### Description

Currently on tc dk-s are ressurected with no runic power.

I might be wrong but coudn't find a video from back then on 3.3.5

### Expected behaviour

Well it seems that on 3.4 dk are ressurected in bg with full runic power.

https://youtu.be/qHdp3z718YI?t=422

### Steps to reproduce the problem

with a 80 dk
.debug bg
queue bg
enter wait for start then die
notice you are ressurected with no runnic power

### Branch

3.3.5

### TC rev. hash/commit

2a587e5ac43f

### Operating system

Windows 10 x64

### Custom changes

None

**Comment by Nyr97** [comment]
Reference to PowerType.db2:

- Only 0x1000 (Set to Max on level up) and 0x2000 (Set to max on initial login) are referenced as to fill on a given hook.
- RUNIC_POWER_COST has no flags whatsoever. Not sure if 0x40 (Not Set to Default on Resurrect) is related to this, as this flag is only in MANA_COST and SOUL_SHARDS_COST. However, every class power is filled completely when resurrected and there's no such flag on any other power.

TL;DR this leads us to believe all power types are supposed to be filled when resurrected, except power points which are built-in combat with given player spells.

**Comment by Ovahlord** [comment]
Check sniffs if there is a full power spell cast. Pets use such spells when being summoned to fill up their power and health.

**Comment by CraftedRO** [comment]
warriors shoudn't be ressurected with full rage: https://youtu.be/CI2_6M7UnEw?t=876

**Comment by acidmanifesto** [comment]
There was alot of bg spamming sniffs in the sniff collective, i would ask the tc devs with access to check with it.

**Comment by Jildor** [comment]
I think it's a mistake to rely on wotlk classic to check certain mechanics specific to 3.3.5

https://www.warcraftmovies.com/movieview.php?id=156233 <-- min 05:50 resurrect and 0 runic power
https://www.youtube.com/watch?v=yUqRXj7wRnc <-- min 2:26 resurrect and 0 runic power

**Comment by CraftedRO** [comment]
That's why I said this in the above issue description:

> I might be wrong but coudn't find a video from back then on 3.3.5

And no, overall is not really a mistake to rely on some wotlk classic stuff to check clearly broked TC 3.3.5 mechanincs (not this case).
